backend/src/services/aws/ec2/ec2.service.ts

```diff
@@ -88,10 +89,21 @@ class EC2 {
     );
 
     const [instance] = data.Instances as Instance[];
-    const { InstanceId: instanceId } = instance;
+    const { InstanceId: instanceId, State: state } = instance;
+    const { Name: stateName } = state as InstanceState;
 
     await this.setInstanceName(instanceId as string, name);
 
+    return {
+      instanceId: instanceId as string,
+      state: stateName as string,
+    };
+  }
+
+  public async getRunningInstanceData(instanceId: string): Promise<{
+    hostname: string;
+    state: string;
```

Comment:
Your method called `getRunningInstanceData` but it returns state. Does it mean that state can be different from 'running'? It will definitely confuse user of this method.